utils/timed_version_test.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTimedVersion(t *testing.T) {
+	t.Run("timed versions are monotonic and comparable", func(t *testing.T) {
+		gen := NewDefaultTimedVersionGenerator()
+		tv1 := gen.New()
+		tv2 := gen.New()
+		tv3 := gen.New()
+
+		assert.True(t, tv3.After(tv1))
```

Comment:
any reason to use assert instead of require here? testify asserts do not terminate the test, but will let execution continue. I've found it to be confusing when tests are failing.

Reply:
i assumed you had a reason for using it in the original tests so i copied it

Reply:
hah I didn't realize I did! probably typed it out due to muscle memory.